cpp/src/table/legacy/table.cpp

```diff
@@ -116,4 +137,22 @@ bool has_nulls(cudf::table const& table) {
   });
 }
 
+table concat_tables(cudf::table const& table1, cudf::table const&table2) {
+    if ((table1.num_columns() == 0) and (table2.num_columns() == 0))
```

Comment:
I don't think these explicit `if` checks are necessary. With the way you construct the vector below, you'll automatically get the same result.

Reply:
This will help me not to come across that exception in table constructor and will also reduce couple of function calls.

Reply:
I'm not concerned about the overhead of a function call. It would greatly simplify the function to remove these. 

I think we should fix the table constructor.

Reply:
ok, I will modify the constructor in the next push.